src/HotChocolate/Core/src/Types/Types/Relay/GlobalIdInputValueFormatter.cs

```diff
@@ -82,6 +101,42 @@ internal class GlobalIdInputValueFormatter : IInputValueFormatter
                 return list;
             }
 
+            if (runtimeValue is IEnumerable<string?> nullableStringEnumerable)
```

Comment:
Seems in testing actually that it could be `is IEnumerable<string?>` and `is IEnumerable<string>` are both always true, which I guess makes sense at runtime since it's the same data structure under the hood, but in practice we then can't distinguish and get the nullability warnings. 

So I could potentially just update the existing block to use `IEnumerable<string?>` and not have two blocks I guess

Reply:
That isn't true for the IdValue struct though, so the block higher seems valid.

```c#
new List<IdValue?>() is IEnumerable<IdValue?>
true
new List<IdValue?>() is IEnumerable<IdValue>
false
new List<string?>() is IEnumerable<string>
true
new List<string?>() is IEnumerable<string?>
true
```